milabench/metadata.py

```diff
@@ -0,0 +1,42 @@
+import os
+from datetime import datetime
+import subprocess
+import cpuinfo
+
+from voir.instruments.gpu import get_gpu_info
+
+
+def _exec(cmd, default):
+    try:
+        return subprocess.check_output(cmd.split(" "), encoding="utf-8").strip()
+    except:
+        return default
+
+
+def machine_metadata():
+    """Retrieve machine metadata"""
+
+    uname = os.uname()
+    gpus = get_gpu_info()
+    cpu = cpuinfo.get_cpu_info()
+
+    return {
+        "cpu": {
+            "count": os.cpu_count(),
+            "brand": cpu.pop("brand_raw", "<unknown>"),
+        },
+        "os": {
+            "sysname": uname.sysname,
+            "nodename": uname.nodename,
+            "release": uname.release,
+            "version": uname.version,
+            "machine": uname.machine,
+        },
+        "accelerators": gpus,
+        "date": datetime.utcnow(),
+        "milabench": {
+            "tag": _exec("git describe --tags", "<tag>"),
+            "commit": _exec("git rev-parse HEAD", "<commit>"),
+            "date": _exec("git show -s --format=%ci", "<date>"),
```

Comment:
That's only going to work when `milabench` is run from the repo directory or a subdirectory, which is going to be the case most of the time, but technically the command can be run from anywhere. Also it won't work unless the package is installed in editable mode on a clone of the repo (otherwise the git data isn't preserved); not much we can do about that, though, so I suppose that's OK.

Reply:
with setuptools we could have populated some version magic in a file.
I am not sure how it could be done for poetry

Reply:
I added a build script that populates a `_version.py` file so all the version info is available even when the git repo is not there.

I am not sure if I should commit the `_version.py` file or not.
We could commit a template and then configure git to ignore subsequent change.